impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -1227,27 +1235,31 @@ private void expireMapping(final K key, final ValueHolder<V> value) {
 
   private OnHeapValueHolder<V> cloneValueHolder(K key, ValueHolder<V> valueHolder, long now, Duration expiration) {
     V realValue = valueHolder.value();
+    boolean veto = checkVeto(CacheStoreHelper.cacheEntry(key, valueHolder, timeSource));
     if(valueCopier instanceof SerializingCopier) {
-      return new SerializedOnHeapValueHolder<V>(valueHolder, realValue, ((SerializingCopier)valueCopier).getSerializer(), now, expiration);
+      return new SerializedOnHeapValueHolder<V>(valueHolder, realValue, veto, ((SerializingCopier)valueCopier).getSerializer(), now, expiration);
     } else {
-      return new CopiedOnHeapValueHolder<V>(valueHolder, realValue, valueCopier, now, expiration);
+      return new CopiedOnHeapValueHolder<V>(valueHolder, realValue, veto, valueCopier, now, expiration);
     }
   }
 
-  private OnHeapValueHolder<V> makeValue(V value, long creationTime, long expirationTime, Copier<V> valueCopier) {
-    if(valueCopier instanceof SerializingCopier) {
-      return makeSerializedValue(value, creationTime, expirationTime, ((SerializingCopier)valueCopier).getSerializer());
+  private OnHeapValueHolder<V> makeValue(K key, V value, long creationTime, long expirationTime, Copier<V> valueCopier) {
+    boolean veto = checkVeto(CacheStoreHelper.cacheEntry(key, value, creationTime, OnHeapValueHolder.TIME_UNIT));
```

Comment:
This creates a cache entry that never has a valid hit rate, while there may be a valid one. Can't we find a way to compute the veto with ALL the information it can use internally?